frontend/src/static/js/main.js

```diff
@@ -104,8 +101,6 @@ window.app = new window.Vue({
 
     // handle opening of sidebar //
     activateTab(tabName) {
-      // changing isTabActive to trigger redrawing of component
-      this.isTabActive = false;
```

Comment:
This one may still be essential to make everything work. A simple workaround to destroy all the component attribute that we want to renew when re-rendering.

Reply:
Oh the tab doesn't rerender anymore in this PR. I'll try removing it and see if it causes any bugs.

Reply:
I got the logic as I tried to do so once. The new way `initialState()` seems to be working.

Reply:
Actually I removed isCollapsed because it wasn't used anywhere after rewriting some stuff. I think it still works.